src/DocumentDB.ChangeFeedProcessor.UnitTests/PartitionManagement/PartitionSynchronizerTests.cs

```diff
@@ -86,5 +87,91 @@ public async Task SplitPartitionAsync_ShouldReturnOnlyNewLeases_IfSplitWasAlread
             Assert.NotNull(result);
             Assert.Equal(new[] { lease20 }, result);
         }
+
+        [Fact]
+        public async Task SplitPartitionAsync_ShouldThrow_IfSinglePartitionAndChildPartitionsLessThan2()
```

Comment:
Let's name it in more specific way, that this is collection provisioned scenario and PKRanges return 1.